packages/@react-stately/collections/src/CollectionBuilder.ts

```diff
@@ -61,6 +61,7 @@ export class CollectionBuilder<T> {
 
   getNode(item: CollectionElement<T>, level: number, value: T, childNodes: Iterable<Node<T>>, parentKey?: Key): Node<T> {
     let key = this.getKey(item, value, parentKey);
+    key = item.props.staticKey || key;
```

Comment:
Generative collections won't/shouldn't have `staticKey`attached to its `<Item>` props so old functionality is preserved. If we are worried about possible React side effects or uniqueness of keys, I could not overwrite `key` here and instead add a `staticKey: item.props.staticKey || key` prop to `node` below and have `TreeCollection` generate its `keyMap` off of `node.staticKey` instead